debian/matrix-synapse-py3.preinst

```diff
@@ -1,31 +0,0 @@
-#!/bin/sh -e
-
-# Attempt to undo some of the braindamage caused by
-# https://github.com/matrix-org/package-synapse-debian/issues/18.
-#
-# Due to reasons [1], the old python2 matrix-synapse package will not stop the
-# service when the package is uninstalled. Our maintainer scripts will do the
-# right thing in terms of ensuring the service is enabled and unmasked, but
-# then do a `systemctl start matrix-synapse`, which of course does nothing -
-# leaving the old (py2) service running.
-#
-# There should normally be no reason for the service to be running during our
-# preinst, so we assume that if it *is* running, it's due to that situation,
-# and stop it.
-#
-# [1] dh_systemd_start doesn't do anything because it sees that there is an
-#     init.d script with the same name, so leaves it to dh_installinit.
-#
-#     dh_installinit doesn't do anything because somebody gave it a --no-start
-#     for unknown reasons.
-
-if [ -x /bin/systemctl ]; then
-    if /bin/systemctl --quiet is-active -- matrix-synapse; then
-        echo >&2 "stopping existing matrix-synapse service"
-        /bin/systemctl stop matrix-synapse || true
-    fi
-fi
-
-#DEBHELPER#
-
-exit 0
```

Comment:
The comment indicates that this is a problem with the py2 synapse package, which is over two years old, so I'd not expect to see upgrades from that any longer.

Give me a minute to re-read how the maintainer scripts work...

Reply:
Looking at the flowcharts in [the Debian policy manual](https://www.debian.org/doc/debian-policy/ap-flowcharts.html), a successful upgrade would invoke:

1. old `prerm upgrade`
2. new `preinst upgrade`
3. old `postrm upgrade`
4. new `postinst configure`

The `prerm` script calls `deb-systemd-invoke stop 'matrix-synapse.service' >/dev/null || true`, so doing the same thing in `preinst` appears to be redundant.

Reply:
> The prerm script calls deb-systemd-invoke stop 'matrix-synapse.service' >/dev/null || true, so doing the same thing in preinst appears to be redundant.

the problem is/was that the prerm in matrix-synapse *didn't* do that (for the reasons set out at lines 16-20).

Reply:
> The comment indicates that this is a problem with the py2 synapse package, which is over two years old, so I'd not expect to see upgrades from that any longer.


I think I'm happy to drop support for upgrades from our own old matrix-synapse package - as you say, it's over two years since that turned into a transitional package.

I also wanted to check that the `matrix-synapse` bug was fixed in downstream debian (since their packaging was forked from our own), but it looks like it was: https://salsa.debian.org/matrix-team/matrix-synapse/-/commit/d11ec18d6b5694e5573c152a3fddda817f6ced50#8756c63497c8dc39f7773438edf53b220c773f67_7_7

In short: fair enough, let's drop this!